### PR TITLE
[api] add ability to filter files that get read

### PR DESCRIPTION
**Pull request by jcrugzz**

### CODE REVIEW

File: workers/webpack.js [hunk 1]
```diff
@@ -97,6 +98,10 @@ function run(next) {
       .once('end', () => done(null, files))
       .on('file', function found(root, file, cb) {
         //
+        // Run the filter function and if it returns false, dont use that file
+        //
+        if (!factory.filter(file)) return void cb();
+        //
```
Comment (3rd-Eden) [inline workers/webpack.js]: OCD: newline

Comment (3rd-Eden) [inline workers/webpack.js]: None blocking ;-)

File: workers/webpack.js [hunk 2]
```diff
@@ -97,6 +98,10 @@ function run(next) {
...
         // Ignore minified files that were found in the directory
         //
         if (file.name.indexOf('.min.') !== -1) return void cb();
```
Comment (indexzero) [inline workers/webpack.js]: Let's not have two different types of filtering:

- Change `opts.filter` to `opts.filters`
- Define `.filter` prototype function that calls all filters
- Make `file.name.indexOf` a default filter.

**Comment by indexzero** [comment]
+1 on green.

**Pull request merged**